bobtask/task.go

```diff
@@ -94,6 +95,18 @@ type Task struct {
 
 	// skippedInputs is a lists of skipped input files
 	skippedInputs []string
+
+	Dependencies []string `yaml:"dependencies"`
+
+	// Contains Dependencies + its bobfile Dependencies
+	AllDependencies []string
+
+	// Paths from /nix/store computed from all its dependencies
+	StorePaths []string
```

Comment:
1:All public members of a bobtask can be written from a Bobfile through Marshal.
2: All public members of a bobtask are consideren by the hash/checksum of this task.

For most of them 1. should be avoided. Looks like we want to consider them at 2. and assure that a different order in dependencies invalidates the hash.